view/frontend/web/js/model/virtualQuote.js

```diff
@@ -0,0 +1,39 @@
+define([
+    'uiComponent',
+    'Magento_Customer/js/customer-data',
+    'ko'
+], function (
+    Component,
+    customerData,
+    ko
+) {
+    'use strict';
+    return Component.extend({
+        defaults: {
+            isVirtual: ko.observable(0).extend({notify: 'always'})
+        },
+
+        getIsVirtual: function () {
+            return this.isVirtual();
+        },
+
+        setIsVirtual: function (isPdp, initExpressResponse = null) {
+            let isVirtual = false;
+
+            if (isPdp && initExpressResponse['is_virtual_quote']) {
+                isVirtual = true;
+            } else if (!isPdp) {
+                const cart = customerData.get('cart');
+                isVirtual = true;
+
+                cart().items.forEach((item) => {
```

Comment:
Looping through the cart items isn't a good way to determine if the quote is virtual or not. For example you can have a `product_type` of `grouped` and the quote could be virtual still due to its children.

Instead we should load the quote model and check if the quote is virtual.

```js
define([
    'uiComponent',
    'Magento_Customer/js/customer-data',
    'Magento_Checkout/js/model/quote',
    'ko'
], function (
    Component,
    customerData,
    quote,
    ko
) {
    'use strict';
    return Component.extend({
        defaults: {
            isVirtual: ko.observable(0).extend({notify: 'always'})
        },

        getIsVirtual: function () {
            return this.isVirtual();
        },

        setIsVirtual: function (isPdp, initExpressResponse = null) {
            let isVirtual = false;

            if (isPdp && initExpressResponse['is_virtual_quote']) {
                isVirtual = true;
            } else if (!isPdp) {
                isVirtual = quote.isVirtual();
            }

            return this.isVirtual(isVirtual);
        }
    });
});

```